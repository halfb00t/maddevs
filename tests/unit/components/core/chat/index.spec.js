import { shallowMount } from '@vue/test-utils'
import { render } from '@testing-library/vue'
import Chat from '@/components/core/chat'
import 'regenerator-runtime'

const stubs = ['ChatIcon', 'ChatModal']
const mockToggleMethod = jest.spyOn(Chat.methods, 'toggleModal')

describe('Chat component', () => {
  it('should render correctly', () => {
    const { container } = render(Chat, { stubs })
    expect(container).toMatchSnapshot()
  })

  it('should toggle the isVisible option by click the button', async () => {
    await shallowMount(Chat, { stubs }).find('button').trigger('click')
    expect(mockToggleMethod).toHaveBeenCalledTimes(1)
  })

  it('isVisible is false by default', () => {
    const wrapper = shallowMount(Chat, { stubs })
    expect(wrapper.vm.isVisible).toBe(false)
  })

  it('Escape sets isVisible to false', () => {
    const wrapper = shallowMount(Chat, {
      stubs,
    })

    const event = {
      key: 'Escape',
    }

    wrapper.vm.isVisible = true
    wrapper.vm.onKeydown(event)

    expect(wrapper.vm.isVisible).toBe(false)
  })
})
