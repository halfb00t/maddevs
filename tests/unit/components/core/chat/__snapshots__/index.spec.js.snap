// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component should render correctly 1`] = `
<div>
  <section
    class="chat"
  >
    <button
      class="chat__button"
    >
      <chaticon-stub />
    </button>
     
    <!---->
  </section>
</div>
`;
